session/payment/session_balance.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Package payment is responsible for ensuring that the consumer can fullfil his obligation to provider.
+// It contains all the orchestration required for value transfer from consumer to provider.
+package payment
+
+import (
+	"errors"
+	"time"
+
+	"github.com/mysteriumnetwork/node/session/balance"
+	"github.com/mysteriumnetwork/node/session/promise"
+)
+
+// BalanceTracker keeps track of current balance
+type BalanceTracker interface {
+	GetBalance() balance.Message
+}
+
+// PromiseValidator validates given promise
+type PromiseValidator interface {
+	Validate(promise.Message) bool
+}
+
+// PeerBalanceSender knows how to send a balance message to the peer
+type PeerBalanceSender interface {
+	Send(balance.Message) error
+}
+
+// ErrPromiseWaitTimeout indicates that we waited for a promise long enough, but with no result
+var ErrPromiseWaitTimeout = errors.New("Did not get a new promise")
```

Comment:
[Error strings should not be capitalized...](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)
```suggestion
var ErrPromiseWaitTimeout = errors.New("did not get a new promise")
```